utils/mil_ros_tools/online_bagger/nodes/online_bagger.py

```diff
@@ -0,0 +1,188 @@
+#!/usr/bin/env python
+
+import resource
+import rospy
+import os
+import rosbag
+import rostopic
+from collections import deque
+import yaml
+import rospkg
+from std_srvs.srv import SetBool
+
+'''
+To call this service call the '/online_bagger' service with the SetBool Parameter set to True.
+Doing this will generate a rosbag with the last x number of seconds saved
+
+for example:
+
+    bagging_server = rospy.ServiceProxy('/online_bagger', SetBool)
+    bag_status = bagging_server(True)
+'''
+
+
+class OnlineBagger(object):
+    def __init__(self):
+        '''
+        - makes dictionary of dequeues
+        - subscribes to set of topics defined by the yaml file in directory
+        - have a streaming option that can be turned on and off in the callback
+        - option to bag data
+        - add a re subscribing option for anything that failed? maybe...?
+        '''
+
+        self.n = 0  # number of iterations
+        self.streaming = True
+        self.dumping = False
+        # self.resubscribe
+
+        self.get_params()
+        self.make_dicts()
+        self.subscribe()
+        rospy.loginfo('subscriber success: %s', self.subscriber_list)
+
+
+        self.bagging_service = rospy.Service('/online_bagger/dump', SetBool, self.start_bagging)
+
+    def get_params(self):
+
+        self.topics = rospy.get_param('/online_bagger/message_topics')
```

Comment:
generally parameters are not in the global name space (starting with /) like this. If you just put the parameter name (like "topics" or "stream_time"), ros will known that you are taking about the param in this node's name space)